mautic/mautic
# Report date fix (mostly added test)

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [y]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This adds test for https://github.com/mautic/mautic/pull/10800 and fixes a small bug when the cell type is empty.

#### Steps to test this PR:


0. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
1. Create either a form or email
2. Submit the form at least once on 3 consecutive days or send the email to any contact(s) on 3 consecutive days. It can be multiple different emails sent to the same contact. It may also help investigation to submit the form or send emails multiple times each day at different times of the day
3. Create a form submissions or emails sent report with customer email address, Name (of the form/email), and Date Submitted/Sent as the columns
4. Save the report, then set date filters for the middle day of the 3-day testing

Expected result: Only data for forms submitted on or emails sent on that middle date should appear in the report
Actual result: Data from the third day of testing will appear, in addition to the middle day.

#### Steps to test this PR:
1. Repeat steps 1-4 from "Steps to reproduce the bug".
2. Only date from the middle day should appear.


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11256"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
